samples/stand-alone-samples/basic_pub_sub/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+if (NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 11)
+endif()
```

Comment:
2 things here:
1) `cmake_minimum_required()` and `project()` should be the very first things in the file, so this should go later.
2) `set()` is the "old" way of doing cmake, where setting a value in one project's file might affect other projects. The "modern cmake" way of doing things is to use functions like `set_target_properties()` where you're setting a specific project's properties. You should probably set c++11 later on in the file, along with other properties, like:
```
set_target_properties(${PUB_SUB_PROJECT_NAME} PROPERTIES
    LINKER_LANGUAGE CXX
    CXX_STANDARD 11)
```